packages/core/src/codewhisperer/service/transformByQ/transformMavenHandler.ts

```diff
@@ -206,3 +204,38 @@ export async function getVersionData() {
     )
     return [localMavenVersion, localJavaVersion]
 }
+
+// run maven 'versions:dependency-updates-aggregate-report' with either 'mvnw.cmd', './mvnw', or 'mvn' (if wrapper exists, we use that, otherwise we use regular 'mvn')
+export function runMavenDependencyUpdateCommands(dependenciesFolder: FolderInfo) {
+    // baseCommand will be one of: '.\mvnw.cmd', './mvnw', 'mvn'
+    const baseCommand = transformByQState.getMavenName() // will be one of: 'mvnw.cmd', './mvnw', 'mvn'
+
+    // Note: IntelliJ runs 'clean' separately from 'install'. Evaluate benefits (if any) of this.
+    const args = [
+        'versions:dependency-updates-aggregate-report',
+        `-DoutputDirectory=${dependenciesFolder.path}`,
+        '-DonlyProjectDependencies=true',
+        '-DdependencyUpdatesReportFormats=xml',
+    ]
+
+    let environment = process.env
+    // if JAVA_HOME not found or not matching project JDK, get user input for it and set here
+    if (transformByQState.getJavaHome() !== undefined) {
+        environment = { ...process.env, JAVA_HOME: transformByQState.getJavaHome() }
+    }
+
+    const spawnResult = spawnSync(baseCommand, args, {
+        // default behavior is looks for pom.xml in this root
+        cwd: dependenciesFolder.path,
+        shell: true,
```

Comment:
are you sure that shell is needed? it can cause side effects depending on the user's shell config. if it's needed, a comment should explain why

Reply:
Historically you and @dhasani23 have talked about this. The most context I have is its needed to help find executable path.

But I have also since questioned if its needed now vs a couple months ago.

Reply:
We have an appsec finding on this and will follow up on this to close by 6/26.